Get flash_attn to compile for CUDA 11.6 linux nightly build (#84941)
pytorch/pytorch: CMakeLists.txt

include(cmake/Dependencies.cmake)

# Moved this cmake set option down here because CMAKE_CUDA_COMPILER_VERSION is not avaialble until now
cmake_dependent_option(
  USE_FLASH_ATTENTION
  "Whether to build the flash_attention kernel for scaled dot product attention" ON
  "USE_CUDA AND NOT ROCM AND NOT MSVC AND NOT CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.6" OFF)
if(USE_FLASH_ATTENTION)
    ADD_DEFINITIONS(-DUSE_FLASH_ATTENTION)
ENDIF()

> 

Reviewer: Nit
```suggestion
```

include(cmake/Dependencies.cmake)

# Moved this cmake set option down here because CMAKE_CUDA_COMPILER_VERSION is not avaialble until now

> option(USE_FLASH_ATTENTION "Whether to build the flash_attention kernel for scaled dot product attention" ON)
> if(MSVC OR (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.6))
>   set(USE_FLASH_ATTENTION OFF)
> ENDIF()

Reviewer: Use `cmake_dependen_option`
```suggestion
cmake_dependent_option(USE_FLASH_ATTENTION "Whether to build the flash_attention kernel for scaled dot product attention" ON "NOT MSVC AND NOT CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11.6" OFF)
```
